interfaces/interfaces.go

```diff
@@ -0,0 +1,16 @@
+package interfaces
```

Comment:
Is there a reason this interface needs to be in its own package? I would expect interfaces to live as close as possible to where they're used - so the top level of this repo alongside client.go and client_test.go

It would be reasonable/expected for this to be in the client.go file alongside the Client type definitions, though keeping it in its own file isnt a problem. 

It's reasonable to put the mocks in their own directory/package though 👍

Reply:
The interfaces are now in the same package as client, had to do the same with the mocks. Had a cirular import dependancy that I couldn't figure out without that in the same package too.